src/actions/tests/test_radio_interface.py

```diff
@@ -0,0 +1,233 @@
+        for i in range(max_retries+2):
+            try:
+                self.rx.acquire_samples(1000, 1000, max_retries)
+            except RuntimeError:
+                # Check if this was the expected error
+                if i == max_retries+1:
+                    self.rx.usrp.reset_bad_acquisitions()
+                    return
+                else:
+                    raise RuntimeError(
+                        (
+                            "RadioInterface failed after {} retries.\r\n" +
+                            "Should have failed at {} retries."
+                        ).format(i-1, max_retries)
+                    )
+        # If no error was thrown, this is also an error
+        raise RuntimeError(
+            "RadioInterface did not throw expected error for\r\n" +
+            "failing to retrieve data {} times in a row.".format(max_retries+1)
+        )
+
+    def test_tune_result_parsing(self):
+        # Use a positive DSP frequency
+        f_lo = 1.0e9
+        f_dsp = 1.0e6
+        self.rx.tune_frequency(f_lo, f_dsp)
+        if not (f_lo == self.rx.lo_freq and
+                f_dsp == self.rx.dsp_freq):
+            raise RuntimeError(
+                (
+                    "Tune result parsing failed.\r\n" +
+                    "Set LO: {}, Parsed LO: {}\r\n" +
+                    "Set DSP: {}, Parsed DSP: {}"
+                ).format(
+                    f_lo, self.rx.lo_freq, f_dsp, self.rx.dsp_freq
+                )
+            )
+
+        # Use a 0Hz for DSP frequency
+        f_lo = 1.0e9
+        f_dsp = 0.0
+        self.rx.frequency = f_lo
+        if not (f_lo == self.rx.lo_freq and
+                f_dsp == self.rx.dsp_freq):
+            raise RuntimeError(
+                (
+                    "Tune result parsing failed.\r\n" +
+                    "Set LO: {}, Parsed LO: {}\r\n" +
+                    "Set DSP: {}, Parsed DSP: {}"
+                ).format(
+                    f_lo, self.rx.lo_freq, f_dsp, self.rx.dsp_freq
+                )
+            )
+
+        # Use a nagative DSP frequency
+        f_lo = 1.0e9
+        f_dsp = -1.0e6
+        self.rx.tune_frequency(f_lo, f_dsp)
+        if not (f_lo == self.rx.lo_freq and
+                f_dsp == self.rx.dsp_freq):
+            raise RuntimeError(
+                (
+                    "Tune result parsing failed.\r\n" +
+                    "Set LO: {}, Parsed LO: {}\r\n" +
+                    "Set DSP: {}, Parsed DSP: {}"
+                ).format(
+                    f_lo, self.rx.lo_freq, f_dsp, self.rx.dsp_freq
+                )
+            )
+
+    def test_load_nonexistent_scale_factor_file(self):
+        errored = False
+        try:
+            self.rx.load_scale_factor_csv('dummy')
+        except IOError:
+            errored = True
+        if not errored:
+            raise RuntimeError(
+                "Did not throw error when loading nonexistent file."
+            )
+        return
```

Comment:
This will be much cleaner with pytest:

```python
def test_load_nonexistent_scale_factor_file():
    msg = "Did not throw error when loading nonexistent file."
    with pytest.raises(IOError, message=msg):
        self.rx.load_scale_factor_csv('dummy')
```

Reply:
Replaced similar instances with `with pytest.raises(Exception,...):`